test/hooks-async.test.js

```diff
@@ -649,3 +649,51 @@ test('The this should be the same of the encapsulation level', async t => {
   await fastify.inject({ method: 'GET', path: '/' })
   await fastify.inject({ method: 'GET', path: '/nested' })
 })
+
+test('preSerializationEnd should handle errors if the serialize method throws', t => {
+  t.test('works with sync preSerialization', t => {
+    t.plan(2)
+    const fastify = Fastify()
+
+    fastify.addHook('preSerialization', (request, reply, payload, next) => {
+      next(null, payload)
+    })
+
+    fastify.post('/', {
+      handler (req, reply) { reply.send({ status: 'not ok' }) },
+      schema: { response: { 200: { status: { type: 'array' } } } }
+    })
+
+    fastify.inject({
+      method: 'POST',
+      url: '/'
+    }, (err, res) => {
+      t.error(err)
+      t.notEqual(res.statusCode, 200)
+    })
+  })
+
+  t.test('works with async preSerialization', t => {
```

Comment:
This test dosn't replicate the issue, it return already:

```
{"status":["n","o","t"," ","o","k"]}
200
```

The `required` was working in your examples

Reply:
but the tests passed, both in the ci and locally O.o
hmm..  i don't mind changing it to whatever you want..
i actually found another bug with `fast-json-stringify` while working on these tests :sweat_smile: 
you can't really use enum in the response schema because the serialize method messes it up

Reply:
while i'm already at it :)

https://github.com/fastify/fast-json-stringify/issues/274